packages/web-components/src/components/fade-in-out/fade-in-out.ts

```diff
@@ -0,0 +1,256 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { property, customElement, LitElement } from 'lit-element';
+import { breakpoints } from '@carbon/layout';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import StableSelectorMixin from '../../globals/mixins/stable-selector';
+import { forEach } from '../../globals/internal/collection-helpers';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Function component that handles fade transition for selected elements.
+ *
+ * @example
+ * import '@carbon/ibmdotcom-styles/scss/internal/scroll-into-view/_scroll-into-view.scss';
+ * import '@carbon/ibmdotcom-web-components/es/components/fade-in-out/fade-in-out.js';
+ *
+ * As an example, the function can be called to target all instances of the
+ * elements in a list:
+ *
+ * const list = 'bx--content-block, bx--content-group';
+ *
+ * For default values of 400ms and 'one and done' play:
+ * <dds-fade-in-out selectorTargets="${selectorTargets}"/>
+ *
+ * With 'continuous play' option:
+ * <dds-fade-in-out selectorTargets="${selectorTargets}" iterations="true"/>
+ *
+ * For custom delay time, set within targeted class in the application's CSS code as such:
+ *
+ * .bx--content-block {
+ *   --#{$dds-prefix}--fade-in-out-delay: 250ms;
+ * }
+ *
+ * @element dds-fade-in-out
+ */
+@customElement(`${ddsPrefix}-fade-in-out`)
+class DDSFadeInOut extends StableSelectorMixin(LitElement) {
+  /**
+   * The CSS selector that selects all fade in-out target elements.
+   */
+  @property({ attribute: 'selectorTargets' })
+  selectorTargets?: string;
+
+  /**
+   * Iteration boolean for continuous play option.
+   *
+   * `true` to keep fade in/out behavior during the entire component lifecycle,
+   * `false` to do it only once.
+   *
+   */
+  @property({ type: Boolean, attribute: 'keep-animation' })
+  keepAnimation = false;
+
+  /**
+   * Amount of columns used for calculation.
+   *
+   * @private
+   */
+
+  private _colSpan = 3;
+
+  /**
+   * Saved list of elements to observe to avoid calling querySelectorAll
+   * more than once.
+   *
+   * @private
+   */
+  private _elements: HTMLElement[] = [];
+
+  /**
+   * Intersection Observer that watches outer viewport.
+   *
+   * @private
+   */
+  private _rootObserver: IntersectionObserver | null = null;
+
+  /**
+   * Intersection observer that watches the inner viewport.
+   *
+   * @private
+   */
+  private _innerObserver: IntersectionObserver | null = null;
+
+  /**
+   * Resize observer to trigger rootMargin recalculations
+   *
+   * @private
+   */
+  private _resizeObserver: any | null = null;
+
+  /**
+   * Intersection Observer options
+   *
+   * @private
+   */
+  private _options = {
+    rootMargin: '0px',
+    threshold: 0,
+  };
```

Comment:
Instead of making `_options` placeholder, it's make `rootMargin` and `threshold` themselves private properties (don't forget adding `_` prefix). In this way, we can track changes separately in future.